### DISCUSSION TITLE
Two trailing newlines in formatted file

### DISCUSSION THREAD

**Discussion by wdscxsj** [a-h/templ#191] [General]
Hi, it seems there are always two trailing newline characters in a formatted `.templ` file, instead of one. Is that intentional? Thanks.

**Comment by a-h** [comment]
No, it's not intentional.

Likely just because of the simple iteration at https://github.com/a-h/templ/blob/6327fb47fae9b52f08c0f92b1978685fe229f95a/parser/v2/types.go#L103-L120

Checking whether it's the last node in the file, and only outputting a single newline would likely fix it.

I'd take a PR for that!

**Comment by wdscxsj** [comment]
Thanks for the tip! PR proposed.

**Comment by wdscxsj** [comment]
By the way, in the `KillAll()` function of `cmd/templ/generatecmd/run/run.go`, is it better to use `cmd.Process.Kill()` instead of TASKKILL and syscall?

**Reply by a-h** [reply]
The reasoning for `KillAll()` is that if you run `go run .` or similar as your task, then `go run` will create a binary in the `tmp` directory, and run that binary. This creates a new process, with its own `pid`.

I originally thought that `process.Kill()` would kill the child processes, but apparently I didn't read the documentation properly. 😂 

So... apparently, the solution is to do this syscall stuff, or use TASKKILL on Windows.

**Reply by wdscxsj** [reply]
Child processes, now I see. Thank you!

**Comment by a-h** [comment]
I think this one is closed...
